src/test/java/com/wavefront/sdk/common/clients/WavefrontClientTest.java

```diff
@@ -64,6 +79,112 @@ private String createString(int size) {
+    }
+
+    @Test
+    void sendSpan() {
+      WavefrontClient wfClient = new WavefrontClient.Builder(mockBackend.baseUrl())
+          .includeSdkMetrics(false)
+          .build();
+      long timestamp = System.currentTimeMillis();
+      UUID traceId = UUID.fromString("01010101-0101-0101-0101-010101010101");
+      UUID spanId = UUID.fromString("00000000-0000-0000-0000-000000000001");
+
+      assertDoesNotThrow(() -> {
+        wfClient.sendSpan("a-name", timestamp, 1138, "a-source", traceId, spanId,
+            null, null, null, null);
+        wfClient.flush();
+      });
+
+      String expectedBody = "\"a-name\" source=\"a-source\" traceId=" + traceId + " spanId=" +
+          spanId + " " + timestamp + " 1138\n";
+      mockBackend.verify(1, postRequestedFor(urlEqualTo("/report?f=trace"))
+          .withRequestBody(matching(expectedBody))
+          .withHeader("Content-Type", WireMock.equalTo("application/octet-stream")));
+    }
+
+    @Test
+    void sendDistribution() {
+      WavefrontClient wfClient = new WavefrontClient.Builder(mockBackend.baseUrl())
+          .includeSdkMetrics(false)
+          .build();
+      long timestamp = System.currentTimeMillis();
+
+      assertDoesNotThrow(() -> {
+        wfClient.sendDistribution("a-name",
+            Collections.singletonList(new Pair<>(1.1, 2)),
+            Collections.singleton(HistogramGranularity.MINUTE),
+            timestamp, "a-source", null);
+        wfClient.flush();
+      });
+
+      String expectedBody = "!M " + timestamp + " #2 1.1 \"a-name\" source=\"a-source\"\n";
+      mockBackend.verify(1, postRequestedFor(urlEqualTo("/report?f=histogram"))
+          .withRequestBody(matching(expectedBody))
+          .withHeader("Content-Type", WireMock.equalTo("application/octet-stream")));
+    }
+
+    @Test
+    void canSendTracesToDifferentPort() {
+      WireMockServer mockTraceBackend = new WireMockServer(wireMockConfig().dynamicPort());
+      mockTraceBackend.stubFor(WireMock.post(urlPathMatching("/report")).willReturn(WireMock.ok()));
+      mockTraceBackend.start();
+      assertNotEquals(mockBackend.port(), mockTraceBackend.port());
+
+      WavefrontClient wfClient = new WavefrontClient.Builder(mockBackend.baseUrl())
+          .tracesPort(mockTraceBackend.port())
+          .includeSdkMetrics(false)
+          .build();
+      long timestamp = System.currentTimeMillis();
+
+      assertDoesNotThrow(() -> {
+        wfClient.sendMetric("a-name", 1.0, timestamp, "a-source", new HashMap<>());
+        wfClient.sendSpan("a-name", timestamp, 1138, "a-source",
+            UUID.fromString("01010101-0101-0101-0101-010101010101"),
+            UUID.fromString("00000000-0000-0000-0000-000000000001"),
+            null, null, null, null);
+        wfClient.flush();
+      });
+
+      mockBackend.verify(1, postRequestedFor(urlEqualTo("/report?f=wavefront")));
+      mockTraceBackend.verify(1, postRequestedFor(urlEqualTo("/report?f=trace")));
```

Comment:
You might consider verifying that traces _aren't_ sent to the metrics backend and vice versa